### DISCUSSION TITLE
Why does GEDCOM 7 spec show HEAD etc as level n and not level 0?

### DISCUSSION THREAD

**Discussion by stevespages** [FamilySearch/GEDCOM#293] [General]
Am I correct in saying that several records including HEAD, INDI and FAM have to be level 0?

If that is not correct please can you give examples where these records would not be level 0.

If it is correct then why does the specification use "n" to denote the level of these records? Wouldn't it be less ambiguous to specify them as level 0?

**Comment by tychonievich** [comment]
> Am I correct in saying that several records including HEAD, INDI and FAM have to be level 0?

Yes, that is correct

> If it is correct then why does the specification use "n" to denote the level of these records?

This notation was introduced in version 5.0. Named rules have their top-level specified as `n` and substructures with `+1`/`+2`/etc. Only the top-level [document](https://gedcom.io/specifications/FamilySearchGEDCOMv7.html#document) production gives absolute levels.

As a result, every structure has a level given only once. The level is set once
```gedstruct
0 <<RECORD>>
```
and is then propagated through various other productions
```gedstruct
n <<INDIVIDUAL_RECORD>>
```
and
```gedstruct
n @XREF:INDI@ INDI                         {1:1}   g7:record-INDI
```
to give the final level of the record.

> Wouldn't it be less ambiguous to specify them as level 0?

There's no technical ambiguity, but doing so might make the presentation a bit simpler. The downsides are fairly minor: if we make the change we'd have the same level specified more than once, and it would change a machine-parseable syntax which could require small patches to tools using it.

**Reply by stevespages** [reply]
I can see it is ultimately not ambiguous so that was the wrong word for me to use. I do think it would make the specification easier for humans to understand. Thanks for clarifying!
